### PR TITLE
SALTO-1461: fixed workato validation error

### PR DESCRIPTION
**Pull request by alonstern**
Fix validation errors in workato

---
The issue was in `resolve`. When a value in an element would reference another value in the same element it would not be fully resolved due to the fact that transformValues creates deep copy of the values

---
_Release Notes_: 
None

### CODE REVIEW

File: packages/workspace/src/expressions.ts [hunk 1]
```diff
@@ -184,13 +184,21 @@ const resolveElement = async (
...
-  const referenceCloner: TransformFunc = ({ value }) => resolveMaybeExpression(
-    value,
-    elementsSource,
-    workingSetElements,
-    undefined,
-    resolveRoot
-  )
+  const referenceCloner: TransformFunc = async ({ value }) => {
+    if (typeof value === 'object') {
+      await Promise.all(Object.keys(value).map(async key => {
```
Comment (roironn) [inline packages/workspace/src/expressions.ts:189]: nit - using a map function to create a side effect it confusing - I'd rather using awu().forEach here...

File: packages/workspace/src/expressions.ts [hunk 2]
```diff
@@ -201,9 +209,6 @@ const resolveElement = async (
   }
 
   const { element } = workingSetElements[elementToResolve.elemID.getFullName()]
-  if (workingSetElements[element.elemID.getFullName()] === undefined) {
```
Comment (roironn) [inline packages/workspace/src/expressions.ts:204]: This lines are used to ensure that an element (which was not in the original elements to resolve) from being resolved twice. Without adding it to the working set - resolve time would be much longer, and cycles are possible. So we can't remove it :)

Why is this removal needed?

**Comment by netama** [comment]
the fact that the validation errors were in workato is not related to the issue, right? can you please change the description to be more indicative about the underlying issue?

**Review by roironn — APPROVED** [review]
